Source/ProgressionSystemRuntime/Private/Components/PSSpotComponent.cpp

```diff
@@ -50,13 +61,55 @@ void UPSSpotComponent::OnUnregister()
 	Super::OnUnregister();
 }
 
+// Called when the end game state was changed to toggle progression widget visibility.
+void UPSSpotComponent::OnGameStateChanged_Implementation(ECurrentGameState CurrentGameState)
+{
+	APlayerCharacter* PlayerCharacter = UMyBlueprintFunctionLibrary::GetLocalPlayerCharacter();
+	if (!ensureMsgf(PlayerCharacter, TEXT("ASSERT: [%i] %s:\n'PlayerCharacter' is not valid!"), __LINE__, *FString(__FUNCTION__)))
+	{
+		return;
+	}
+	const FPlayerTag& PlayerTag = PlayerCharacter->GetPlayerTag();
+
+	if (GetMeshChecked().GetPlayerTag() == PlayerTag)
+	{
+		constexpr bool bApplySkin = true;
+
+		switch (CurrentGameState)
+		{
+		case ECurrentGameState::Menu:
+			RefreshAmountOfUnlockedSkins(bApplySkin);
+			break;
+		default: break;
+		}
+	}
+}
+
+void UPSSpotComponent::OnCurrentScoreChanged_Implementation(const FPSSaveToDiskData& CurrentSaveToDiskDataRow, const FPSRowData& CurrenProgressionSettingsRow)
+{
+	APlayerCharacter* PlayerCharacter = UMyBlueprintFunctionLibrary::GetLocalPlayerCharacter();
+
+	if (!ensureMsgf(PlayerCharacter, TEXT("ASSERT: [%i] %s:\n'PlayerCharacter' is not valid!"), __LINE__, *FString(__FUNCTION__)))
```

Comment:
Per my previous review, please dont use `*FString(__FUNCTION__)` anymore, but `__FUNCTION__` along with `%hs` for any new code

Reply:
did a bit more cleaning in projecty https://github.com/moinm3uw/ProgressionSystem/pull/59/commits/4b48f10c8cf09fe5972ce1e7cc9f512936f18b9b